### PR TITLE
I terminus

### PR DESCRIPTION
**Pull request by peersky**
## Related issues
#13

### CODE REVIEW

**Comment by peersky** [comment]
```
cd build/contracts
diff  <(jq --sort-keys .abi TerminusFacet.json) <(jq --sort-keys .abi ITerminus.json)        
3,7d2
<     "inputs": [],
<     "stateMutability": "nonpayable",
<     "type": "constructor"
<   },
<   {
743c738
<         "name": "poolID",
---
>         "name": "id",
(.dao) t@Peersky-mb contracts % diff  <(jq --sort-keys .abi ERC1155WithTerminusStorage.json) <(jq --sort-keys .abi IERC1155WithTerminusStorage.json) 
3,7d2
<     "inputs": [],
<     "stateMutability": "nonpayable",
<     "type": "constructor"
<   },
<   {
346c341
<         "name": "poolID",
---
>         "name": "id",
```

File: contracts/interfaces/ITerminus.sol [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+import "./IERC1155WithTerminusStorage.sol";
+
+interface ITerminus is IERC1155WithTerminusStorage {
```
Comment (zomglings) [inline contracts/interfaces/ITerminus.sol:6]: I would rather this was codegen'd instead of created manually.

**Review by zomglings — CHANGES_REQUESTED** [review]
Terminus interface has changed in the most recent releases. This needs to be updated.

Given that Terminus is still in flux, I would rather do this kind of work using codegen than manually maintaining `ITerminus.sol`. @peersky : Do you want to build the code generator?

It would also be nice to have a tool which could test an interface against an actual ABI to make sure that the interface is complete (or as complete as it needs to be).

**Comment by zomglings** [comment]
Tests are passing on my machine.

**Comment by peersky** [comment]
@zomglings there already is hh plugin that does such generation: https://github.com/dmihal/hardhat-interface-generator if we plan to migrate to hardhat than we can use it, otherwise script is quite simple to reapply to our realities

Nevertheless, in current PR there are points where source code has to be modified anyway so purely generating interface file is not enough.

**Comment by zomglings** [comment]
Stale.

**Pull request closed without merging**